File: src/ProjectR/content/scripts/afflictions/Passives/RockHard.lua
function GetName()
  return "Rock Hard"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { TakingDamage }
end

function OnAttachment(char)
end

function OnRemoval(char)
end

function Trigger(hookPoint, char)
  if hookPoint == TakingDamage and GetCurrentSpell():GetSpellType() == Physical and Roll(0, 99) < 20 then
    param1 = 0
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/RockHard.cs
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;

namespace ProjectR.Scripting.Afflictions
{
    public class RockHard : Affliction
    {
        public override string Name { get { return "Rock Hard"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.TakingDamage, 
                };
            }
        }

        protected override void OnTakingDamage(ICharacter character, ref double damage)
        {
            if (GetCurrentSpell().SpellType == SpellType.Physical && RollPercentage(20))
            {
                damage = 0;
            }
        }
    }
}